### PR TITLE
Client mutual auth

### PR DESCRIPTION
**Pull request by alexmgr**
Adds client authentication support:
- Adds missing TLSCertificateRequest and modifies TLSCertificateVerify
- Adds some helper functions to TLSSessionCtx to get a signed hash
- An client side mutual authentication example

### CODE REVIEW

**Comment by alexmgr** [comment]
Ah, didn't see you just merged #57 . Let me rebase on top of that after committing examples.

File: scapy_ssl_tls/ssl_tls.py [hunk 1]
```diff
@@ -594,18 +594,46 @@ class TLSCertificateList(PacketNoPayload):
...
     name = "TLS Certificate Verify"
-    fields_desc = [
-                   PacketField("digitally-signed", None, TLSDigitallySigned),
-                  ]  
+    fields_desc = [PacketField("alg", None, TLSSignatureHashAlgorithm),
+                   XFieldLenField("sig_length", None, length_of="sig", fmt="H"),  # ASN.1 signature element
+                   StrLenField("sig", "", length_from=lambda x:x.sig_length)]
+
+
+TLS_CERTIFICATE_TYPE = {0x1: "rsa_sign",
```
Comment (tintinweb) [inline scapy_ssl_tls/ssl_tls.py]: There is an enum with the shiny autogenerated name `TLS_CLIENTCERTIFICATETYPE_IDENTIFIERS_REGISTRY` in `ssl_tls_registry`. If you take this one then it'll be auto updated with the registry_fetch util.

    TLS_CERTIFICATE_TYPE = registry.TLS_CLIENTCERTIFICATETYPE_IDENTIFIERS_REGISTRY
    TLSCertificateType = EnumStruct(TLS_CERTIFICATE_TYPE)

Comment (alexmgr) [inline scapy_ssl_tls/ssl_tls.py]: Perfect. Thought about doing this the proper way, then forgot ;).
Agreed, way to go. Let me modify that.

**Comment by tintinweb** [comment]
:+1: 
// checked client_rsa_mutual_auth.py against s_client -Verify  // scapy 2.3.2, openssl 1.0.1

    #> python examples/client_rsa_mutual_auth.py localhost 4433
    #> openssl s_server -Verify 5 -CAfile scapy-ssl_tls/tests/integration/keys/scapy-tls-ca.crt.pem
    verify depth is 5, must return a certificate
    Using default temp DH parameters
    Using default temp ECDH parameters
    ACCEPT
    depth=1 C = US, ST = Some-State, O = Some-Org, CN = scapy-tls-ca
    verify return:1
    depth=0 C = US, ST = Some-State, O = Some-Org, CN = test-user, emailAddress = test-user@scapy.tls
    verify return:1
    ...
    Client certificate
    ...
    subject=/C=US/ST=Some-State/O=Some-Org/CN=test-user/emailAddress=test-user@scapy.tls
    issuer=/C=US/ST=Some-State/O=Some-Org/CN=scapy-tls-ca
    Shared ciphers:ECDHE-RSA-AES128-SHA256
    CIPHER is ECDHE-RSA-AES128-SHA256
    Secure Renegotiation IS NOT supported
    GET / HTTP/1.1
    HOST: localhost

    ERROR
    shutting down SSL
    CONNECTION CLOSED
    ACCEPT

**Pull request merged**